corehq/apps/users/models.py

```diff
@@ -2268,17 +2272,24 @@ def get_dimagi_emails_by_domain(cls, domain):
     def get_location_id(self, domain):
         return getattr(self.get_domain_membership(domain), 'location_id', None)
 
+    def _get_location_or_clear(self, domain, db_get_call):
+        loc_id = self.get_location_id(domain)
+        if loc_id:
+            try:
+                return db_get_call(loc_id)
+            except (ResourceNotFound, SQLLocation.DoesNotExist):
+                self.get_domain_membership(domain).location_id = None
+        return None
+
     @memoized
     def get_sql_location(self, domain):
```

Comment:
same comment as above - though I think easier to handle in this case because you can just name these functions to `get_or_clear` or `get_or_unset`